automerge-protocol/src/lib.rs

```diff
@@ -400,7 +408,7 @@ pub enum Diff {
 pub struct MapDiff {
     pub object_id: ObjectId,
     #[serde(rename = "type")]
-    pub obj_type: MapType,
+    pub map_type: MapType,
```

Comment:
Rather than having these `map_type` and `seq_type` tags, would it be worth also splitting the diffs up. E.g `MapDiff`, `TableDiff`, `ListDiff`, `TextDiff`?

Reply:
Ah, I was thinking about doing this but stopped short. I'll follow through with this now.

Reply:
Pushed